### DISCUSSION TITLE
sagex/api not found (re: Home Assistant)

### DISCUSSION THREAD

**Discussion by stanger89** [OpenSageTV/Community#108] [Other Plugins]
Hi, been forever since I really had to do anything with my SageTV setup, but I'm trying to get it setup with Home Assistant (I'm getting a new IP based remote system).  I'm having a bit of trouble though.  There's a custom component for HA (https://github.com/jdeath/SageTVHomeassistant) that uses sagex, however I when I try to access the sagex/api I get a 404 error.  The plugin shows as installed in the SageTV plugin manager and there's nothing obvious in my log file.  Any ideas what I should check for?

**Comment by jusjoken** [comment]
Are you using the latest sagex plugin and Jetty?

Also check in the logs folder and upload any relevant logs.

Ken

**Reply by stanger89** [reply]
Yeah, I made sure those are updated.  The webserver is working fine, scratch that, actually now after updating those the webserver shows as not found too.  Here's the sagetv_0.log, Thanks!
[sagetv_0.zip](https://github.com/OpenSageTV/Community/files/13837267/sagetv_0.zip)

**Comment by wayner9** [comment]
What happens when you try to go to the sagex landing page of the web UI?
This would be at http://192.168.1.245:8080/sagex/ assuming the SageTV server is at IP 192.168.1.245 and it uses port 8080.  
Or even the regular webserver landing page of http://192.168.1.245:8080/sage/Home

Thanks for posting about this - I didn't know that Home Assistant had anything for SageTV.  I have started playing around with HA and wouldn't mind trying this out.

**Comment by stanger89** [comment]
I get a 404 error from Jetty:

HTTP ERROR 404 Not Found
URI:	/sagex
STATUS:	404
MESSAGE:	Not Found
SERVLET:	-
Powered by Jetty:// 9.4.44.v20210927


Seems like I'm probably missing some files for some reason?

**Reply by wayner9** [reply]
It looks like your Sagex got hosed somehow.  Do all of your other SageTV web pages work?  Possibly it is a Jetty error rather than Sagex.

If it seems to just be SageX then maybe delete and reinstall Sagex?  I think the plugin is called SageTV Sagex Remote API Services (sagex-services).  Or upgrade it if there is an upgrade.  I am personally running older versions of 9.1.7 and 9.2.6.1 on my two main servers and there is a newer version of 9.2.8.1 from 6/1/22.

**Reply by stanger89** [reply]
Yeah, unfortunately I don't think you can just uninstall it, since other stuff depends on it.

**Comment by jusjoken** [comment]
Your log shows the following plugin library failed...

PLULOADEDGINS=[SageTVPlugin[commons-net state=Failed name=commons-net
version=1.4.2

Can you check what plugin manager shows for this..

**Comment by stanger89** [comment]
You looking for this?
![image](https://github.com/OpenSageTV/Community/assets/18076657/75d4c2f9-899c-4224-9e76-72e953d1a047)

**Comment by jusjoken** [comment]
Yes...so not failed. Hmm.  I will review again in the mo

**Comment by stanger89** [comment]
Yeah it's weird, I went through and updated all my out of date plugins, and looking at the log afterwards, it seems a whole bunch of plugins show as "state=Failed", however in the plugin manager in placeshifter none show as failed.   I'm kind of tempted to just uninstall all the plugins and reinstall them.   I've probably got more than I really use at this point.
